napari/layers/image/image.py

```diff
@@ -475,51 +475,43 @@ def _set_view_slice(self):
             for d in self.dims.displayed:
                 scale[d] = self.level_downsamples[self.data_level][d]
             self._transforms['tile2data'].scale = scale
 
-            if np.any(disp_shape > self._max_tile_shape):
-                for d in self.dims.displayed:
-                    indices[d] = slice(
-                        self._top_left[d],
-                        self._top_left[d] + self._max_tile_shape,
-                        1,
-                    )
-                # Note that top left marks the location of top left canvas
-                # pixel in data coordinates
-                self._transforms['tile2data'].translate = (
-                    self._top_left
-                    * self._transforms['data2world'].scale
-                    * self._transforms['tile2data'].scale
+            for d in self.dims.displayed:
+                indices[d] = slice(
+                    self.corner_pixels[0, d], self.corner_pixels[1, d] + 1, 1,
                 )
-            else:
-                self._transforms['tile2data'].translate = np.zeros(self.ndim)
+            self._transforms['tile2data'].translate = (
+                self.corner_pixels[0]
+                * self._transforms['data2world'].scale
+                * self._transforms['tile2data'].scale
+            )
 
             image = np.asarray(
                 self._data_pyramid[level][tuple(indices)]
             ).transpose(order)
 
-            if level == len(self._data_pyramid) - 1:
-                thumbnail = image
-            else:
-                # Slice thumbnail
-                indices = np.array(self.dims.indices)
-                downsampled_indices = (
-                    indices[not_disp] / self.level_downsamples[-1, not_disp]
-                )
-                downsampled_indices = np.round(
-                    downsampled_indices.astype(float)
-                ).astype(int)
-                downsampled_indices = np.clip(
-                    downsampled_indices, 0, self.level_shapes[-1, not_disp] - 1
-                )
-                indices[not_disp] = downsampled_indices
-                thumbnail = np.asarray(
-                    self._data_pyramid[-1][tuple(indices)]
-                ).transpose(order)
+            if np.product(image.shape) == 0:
+                image = np.zeros((1, 1))
+
+            # Slice thumbnail
+            indices = np.array(self.dims.indices)
+            downsampled_indices = (
+                indices[not_disp] / self.level_downsamples[-1, not_disp]
+            )
+            downsampled_indices = np.round(
+                downsampled_indices.astype(float)
+            ).astype(int)
+            downsampled_indices = np.clip(
+                downsampled_indices, 0, self.level_shapes[-1, not_disp] - 1
+            )
+            indices[not_disp] = downsampled_indices
+            thumbnail = np.asarray(
+                self._data_pyramid[-1][tuple(indices)]
+            ).transpose(order)
```

Comment:
I thought thumbnails had a standard size (32x32)? If so, then could we rename this e.g. `thumbnail_source`?

Reply:
good point, renamed `thumbnail_source`